perses/rjmc/topology_proposal.py

```diff
@@ -1470,6 +1474,98 @@ def generate_oemol_from_pdb_template(pdbfile):
+
+        new_rescharge, current_rescharge = 0,0
+        resname_to_charge = {current_resname: 0, new_resname: 0}
+        for resname in [new_resname, current_resname]:
+            if resname in PolymerProposalEngine._negative_aminos:
+                resname_to_charge[resname] -= 1
+            elif resname in PolymerProposalEngine._positive_aminos:
+                resname_to_charge[resname] += 1
+
+        return resname_to_charge[new_resname] - resname_to_charge[current_resname]
+
+    @staticmethod
+    def get_counterion_indices(charge_diff,
+                               old_res,
+                               new_res,
+                               new_positions,
+                               new_topology,
+                               positive_ion_name='NA',
+                               negative_ion_name='CL',
+                               radius=0.3):
+        """
+        return a randomized counterion index to neutralize (index w.r.t. new_topology)
+
+        Parameters
+        ----------
+        charge_diff : int
+            the charge difference between the new_system - old_system
+        old_res : openmm.Topology.residue
+            old mutating residue
+        new_res : openmm.Topology.residue
+            new mutating residue
+        new_positions : np.ndarray(N, 3)
+            positions of atoms corresponding to new_topology
+        new_topology : openmm.Topology
+            topology of new system (with new_res)
+        positive_ion_name : str, default 'NA'
+            name of positive ions to identify as candidates for neutralization (mdtraj naming convention)
+        negative_ion_name : str, default 'CL'
+            name of negative ions to identify as candidates for neutralization (mdtraj naming convention)
+        radius : float, default 0.3
+            minimum distance (in nm) that all candidate ions must be from 'protein atoms'
+
+        Returns
+        -------
+        ion_idx : np.array(abs(charge_diff))
+            index of counterion(s) that are identified for neutralization
+        """
+
+        import mdtraj as md
+        # Create trajectory
+        traj = md.Trajectory(new_positions[np.newaxis, ...], md.Topology.from_openmm(new_topology))
+
+        # Get ion atom indices
+        if charge_diff < 0:
+            ion_name = negative_ion_name
+        elif charge_diff > 0:
+            ion_name = positive_ion_name
+
+        atoms = list(traj.topology.select(f"resname {ion_name}"))
+
+        query_indices = traj.top.select('protein')
+
+        # Get ion indices within radius of mutated residue
+        neighboring_ions = list(md.compute_neighbors(traj, radius, query_indices, haystack_indices=atoms))
+
+        # Get ion indices outside of radius of mutated residue
+        nonneighboring_ions = [atom for atom in atoms if atom not in neighboring_ions]
+        assert len(nonneighboring_ions) > 0, f"there are no available nonneighboring ions"
```

Comment:
Did you mean to have some more debugging information here? If not, I think you should drop the f-string formatting

Reply:
Fixed this!